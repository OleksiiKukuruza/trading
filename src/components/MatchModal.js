import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import styled from 'styled-components';
import { Cell, HeaderRow, Row } from './Table';
import { matchPropType } from './propTypes';

export const StyledBlock = styled.div`
  border: 1px solid ${props => props.theme.primary.main};

  :not(:last-child) {
    margin-bottom: ${props => 4 * props.theme.spacingUnit}px;
  }
`;

const MatchModal = ({ match, onClose }) => (
  <Modal
    isOpen
    onRequestClose={onClose}
    style={{
      content: {
        bottom: 'auto',
        minHeight: '10rem',
        left: '50%',
        padding: '2rem',
        position: 'fixed',
        right: 'auto',
        top: '50%',
        transform: 'translate(-50%,-50%)',
        minWidth: '20rem',
        width: '80%',
        maxWidth: '60rem'
      }
    }}
  >
    <StyledBlock>
      <HeaderRow>
        <Cell>General Info:</Cell>
      </HeaderRow>
      <Row>
        <Cell>Match Time</Cell>
        <Cell>{new Date(match.time).toLocaleTimeString()}</Cell>
      </Row>
      <Row>
        <Cell>Match Price</Cell>
        <Cell>{(match.buy.price + match.sell.price) / 2}</Cell>
      </Row>
      <Row>
        <Cell>Match Amount</Cell>
        <Cell>{Math.min(match.sell.quantity, match.buy.quantity)}</Cell>
      </Row>
    </StyledBlock>
    <StyledBlock>
      <HeaderRow>
        <Cell>Sell Info:</Cell>
      </HeaderRow>
      <Row>
        <Cell>Id:</Cell>
        <Cell>{match.sell.id}</Cell>
      </Row>
      <Row>
        <Cell>Price:</Cell>
        <Cell>{match.sell.price}</Cell>
      </Row>
      <Row>
        <Cell>Amount:</Cell>
        <Cell>{match.sell.quantity}</Cell>
      </Row>
    </StyledBlock>
    <StyledBlock>
      <HeaderRow>
        <Cell>Buy Info:</Cell>
      </HeaderRow>
      <Row>
        <Cell>Id:</Cell>
        <Cell>{match.buy.id}</Cell>
      </Row>
      <Row>
        <Cell>Price:</Cell>
        <Cell>{match.buy.price}</Cell>
      </Row>
      <Row>
        <Cell>Amount:</Cell>
        <Cell>{match.buy.quantity}</Cell>
      </Row>
    </StyledBlock>
  </Modal>
);

MatchModal.propTypes = {
  match: matchPropType.isRequired,
  onClose: PropTypes.func.isRequired
};

export default MatchModal;
