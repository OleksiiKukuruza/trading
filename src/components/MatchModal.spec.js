import React from 'react';
import { shallow } from 'enzyme';
import MatchModal, { StyledBlock } from './MatchModal';
import theme from '../theme';

const setup = (Component, props) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('MatchModal', () => {
  it('renders without crashing', () => {
    const { wrapper } = setup(MatchModal, {
      onClose: () => null,
      match: {
        time: 123,
        sell: { id: 3, price: 50, quantity: 3 },
        buy: { id: 4, price: 60, quantity: 6 }
      }
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledBlock', () => {
    const { wrapper } = setup(StyledBlock, { theme });
    expect(wrapper).toMatchSnapshot();
  });
});
