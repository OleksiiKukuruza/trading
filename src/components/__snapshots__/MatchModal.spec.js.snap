// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MatchModal renders StyledBlock 1`] = `
.c0 {
  border: 1px solid #54b6c8;
}

.c0:not(:last-child) {
  margin-bottom: 32px;
}

<div
  className="c0"
/>
`;

exports[`MatchModal renders without crashing 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "bottom": "auto",
        "left": "50%",
        "maxWidth": "60rem",
        "minHeight": "10rem",
        "minWidth": "20rem",
        "padding": "2rem",
        "position": "fixed",
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%,-50%)",
        "width": "80%",
      },
    }
  }
>
  <styled.div>
    <Styled(styled.div)>
      <styled.div>
        General Info:
      </styled.div>
    </Styled(styled.div)>
    <styled.div>
      <styled.div>
        Match Time
      </styled.div>
      <styled.div>
        3:00:00 AM
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        Match Price
      </styled.div>
      <styled.div>
        55
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        Match Amount
      </styled.div>
      <styled.div>
        3
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <Styled(styled.div)>
      <styled.div>
        Sell Info:
      </styled.div>
    </Styled(styled.div)>
    <styled.div>
      <styled.div>
        Id:
      </styled.div>
      <styled.div>
        3
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        Price:
      </styled.div>
      <styled.div>
        50
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        Amount:
      </styled.div>
      <styled.div>
        3
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <Styled(styled.div)>
      <styled.div>
        Buy Info:
      </styled.div>
    </Styled(styled.div)>
    <styled.div>
      <styled.div>
        Id:
      </styled.div>
      <styled.div>
        4
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        Price:
      </styled.div>
      <styled.div>
        60
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        Amount:
      </styled.div>
      <styled.div>
        6
      </styled.div>
    </styled.div>
  </styled.div>
</Modal>
`;
